### PR TITLE
update deprecated calls

### PR DESCRIPTION
**Pull request by meara**
Noticed these while doing some Rails 5 prep/gem upgrades. Seemed worth cleaning up.

Doing https://pm.tablexi.com/issues/137255 as well, while I'm at it.

### CODE REVIEW

File: app/views/facility_journals/_problem_transactions.html.haml [hunk 1]
```diff
@@ -31,7 +31,7 @@
             %td= format_usa_date(order_detail.public_send(:"#{@date_range_field}"))
             %td.user-order-detail.order-note
               .order-detail-description
-                = order_detail_description(order_detail)
+                = OrderDetailPresenter.new(order_detail).description_as_html
                 - order_detail.extend(PriceDisplayment)
```
Comment (jhanggi) [inline app/views/facility_journals/_problem_transactions.html.haml]: Should we kill this `PriceDisplayment` while we're here?

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -18,21 +18,19 @@ def html_title(title = nil)
     (full_title + app_name).html_safe
   end
 
-  def order_detail_description(order_detail) # TODO: deprecate in favor of OrderDetailPresenter#description_as_html
+  def order_detail_description(order_detail)
```
Comment (jhanggi) [inline app/helpers/application_helper.rb]: If there aren't anymore of these left, should we get rid of the methods?

Comment (meara) [inline app/helpers/application_helper.rb]: Not sure if there might be any in school specific engines, which is why I've got the deprecation warnings here.

Comment (jhanggi) [inline app/helpers/application_helper.rb]: 👍

File: app/presenters/order_detail_presenter.rb [hunk 3]
```diff
@@ -18,6 +18,15 @@ def description_as_html
...
   end
 
+  def description_as_text
+    name = product.to_s
+    if bundle
+      name.prepend("#{bundle} -- ")
+    else
+      name
+    end.html_safe
+  end
```
Comment (waggles) [inline app/presenters/order_detail_presenter.rb]: this saves duplication but results in some slightly-too-complex code paths, what do you think about flattening some of the logic?

Comment (meara) [inline app/presenters/order_detail_presenter.rb]: You're right this is not the most beautiful method. I basically copy and pasted this from the application_helper, don't feel super motivated to change it now. 😜

File: app/views/notifier/order_receipt.html.haml [hunk 4]
```diff
@@ -35,8 +35,7 @@
       %th= OrderDetail.human_attribute_name("#{label_key_prefix}_total")
       %th= OrderDetail.human_attribute_name(:note).pluralize
   %tbody
-    - @order.order_details.each do |order_detail|
-      - order_detail.send(:extend, PriceDisplayment)
+    - OrderDetailPresenter.wrap(@order.order_details).each do |order_detail|
```
Comment (jhanggi) [inline app/views/notifier/order_receipt.html.haml]: Looks like we're double wrapping in line 42 now.

File: app/views/orders/receipt.html.haml [hunk 5]
```diff
@@ -31,8 +31,7 @@
         %th.currency= t('.th.adjust')
       %th.currency= t('.th.total')
   %tbody
-    - @order_details.each do |order_detail|
-      - order_detail.send(:extend, PriceDisplayment)
+    - OrderDetailPresenter.wrap(@order_details).each do |order_detail|
```
Comment (jhanggi) [inline app/views/orders/receipt.html.haml]: Doing some double wrapping in line 40.

File: app/models/reports/account_transactions_report.rb [hunk 6]
```diff
@@ -72,14 +72,14 @@ def headers
...
-    order_detail.extend(PriceDisplayment)
+    order_detail = OrderDetailPresenter.new(order_detail)
 
     [
       order_detail.order.id,
       order_detail.id,
       format_usa_date(order_detail.send(:"#{@date_range_field}")),
       order_detail.order.facility,
-      order_detail_description(order_detail),
+      OrderDetailPresenter.new(order_detail).description_as_html,
```
Comment (jhanggi) [inline app/models/reports/account_transactions_report.rb]: Is this double wrapping?

**Comment by meara** [comment]
Updated. Just checked again for `can?` calls anywhere within the blocks and we should be good.

**Pull request merged**